warehouse/query-core/src/main/java/datawave/query/index/lookup/RangeStream.java

```diff
@@ -809,15 +818,101 @@ public Object visit(ASTAssignment node, Object data) {
         if (Constants.SHARD_DAY_HINT.equals(identifier)) {
             JexlNode myNode = JexlNodeFactory.createExpression(node);
             String[] shardsAndDays = StringUtils.split(JexlASTHelper.getLiteralValue(node).toString(), ',');
-            if (shardsAndDays.length > 0) {
-                return ScannerStream.withData(createIndexScanList(shardsAndDays).iterator(), myNode);
-            } else {
+
+            if (shardsAndDays.length == 0) {
                 return ScannerStream.noData(myNode);
             }
+
+            // it is important that we check for a day range. in that case we need to build a different iterator
+            // to preserve search parallelism.
+            boolean hintContainsDays = checkHintForDays(shardsAndDays);
+            if (hintContainsDays) {
```

Comment:
A little confused why we can't simply use the HintToShardIterator in any case?  How would that deny us parallelism ?

Reply:
If the hint or hints are all shards then we can use the existing mechanism in the range stream. This only uses the complex logic if it's necessary.